postgres/src/sqlx/test_utils.rs

```diff
@@ -0,0 +1,57 @@
+use crate::sqlx::options::PgDatabaseOptions;
+use sqlx::{Connection, Executor, PgConnection, PgPool};
+
+/// Creates a new PostgreSQL database and returns a connection pool to it.
+///
+/// Establishes a connection to the PostgreSQL server using the provided options,
+/// creates a new database, and returns a [`PgPool`] connected to the new database.
+/// Panics if the connection fails or if database creation fails.
+pub async fn create_pg_database(options: &PgDatabaseOptions) -> PgPool {
+    // Create the database via a single connection.
+    let mut connection = PgConnection::connect_with(&options.without_db())
+        .await
+        .expect("Failed to connect to Postgres");
+    connection
+        .execute(&*format!(r#"CREATE DATABASE "{}";"#, options.name))
+        .await
+        .expect("Failed to create database");
+
+    // Create a connection pool to the database and run the migration.
```

Comment:
Nit: migrations don't run here as per the comment.

Reply:
Thanks! I left it from a previous impl.